### DISCUSSION TITLE
Fr Realearn Midi transformation : Polyaftertouch to mpe and vice versa

### DISCUSSION THREAD

**Discussion by ElVicente** [helgoboss/helgobox#1196] [Ideas]
I have two use cases in mind : polyaftertouch midi editing is a nightmare currently in reaper, so being able to edit those in mpe channel pressure (with distinct reallocated midi channels for each notes) would be so great for those softsynths witch support polyat but not mpe.

The other one would be to use my synthstrom deluge to sequence my prophet2.
The deluge only can send mpe midi, but the rev2 can only receive polyat. Using realearn to do the translation would be so cool.

What do you think? Would that be something worth the implementation time needed ?

**Comment by helgoboss** [comment]
Doesn't MPE just use normal MIDI messages? If that's the case, then maybe all you need to find out is what "MPE channel pressure" actually is - in terms of MIDI messages. And then you can maybe use mappings with source "MIDI: Polyphonic after touch" and target "MIDI: Send message" to do the translation already now?

**Reply by ElVicente** [reply]
Thanks for the reply.
Mpe is pretty much hacking regular midi 1.0 
Mpe channel pressure is just exactly regular aftertouch (not polyphonic)
But each note got his own midi channel among the note channels.
So each note got his unique pressure, pitch bend and timbre (cc74) value.

But there is a master channel (channel 1 or 16) witch keep the informations you want to keep for global sound modlations such as mod wheel.

**Reply by ElVicente** [reply]
Would it be possible with each note to cycle through a list of realearn unit or group ?

**Reply by helgoboss** [reply]
Ah yes, and the channel is somehow assigned when pressing the note, right? Well, that sort of logic is a bit beyond what ReaLearn mappings support.

However, it could easily be done with JSFX. I guess you were thinking of ReaLearn because it can modify incoming MIDI globally, right?

**Reply by ElVicente** [reply]
Yep that's it, ok.
I thinked at realearn because I manage to pretty much use it :)
JSFX feels beyond my reach since my college years of programming school are so far away now. I think I'll try to get to it when I have more time...

Does the opposite transformation would be in the reach what Realearn can do ? 
From mpe notes with distinct midi channels and channel pressure to midi channel 1 and Polyaftertouch ?

**Reply by helgoboss** [reply]
I suppose the opposite would be possible.

**Comment by ElVicente** [comment]
Well it turns out that reaper support polyat editing very well now.
So no need for any of this now.
 :)
